sdk/sdkintegrations/initconn.go

```diff
@@ -0,0 +1,35 @@
+package sdkintegrations
+
+import (
+	"fmt"
+	"net/http"
+
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+// FinalizeConnectionInit finalizes the connection initialization.
+// This is done by encoding the init data into the connection initialization URL
+// and redirecting the user there.
+func FinalizeConnectionInit(w http.ResponseWriter, r *http.Request, iid sdktypes.IntegrationID, data []sdktypes.Var) {
+	vars, err := kittehs.EncodeURLData(data)
+	if err != nil {
+		http.Error(w, "Failed to encode URL vars", http.StatusInternalServerError)
+		return
+	}
+
+	cid, err := sdktypes.ParseConnectionID(r.URL.Query().Get("cid"))
+	if err != nil {
+		http.Error(w, "Failed to parse connection ID", http.StatusBadRequest)
+		return
+	}
+
+	id := cid.String()
```

Comment:
- when cid.String() is ""?
- if its "" then id is integration name, right? will redirect to /connections/<integration_name> work?

Reply:
> * when cid.String() is ""?

when cid is invalid, meaning not supplied in request. `ParseConnectionID` will return `InvalidConnectionID, nil` in that case.

> * if its "" then id is integration name, right? will redirect to /connections/<integration_name> work?

no, it'll still redir to connections, but `initConnection` in https://github.com/autokitteh/autokitteh/blob/489ce1df324f2f11aa5735ca7e7b935500fb9ae6/internal/backend/dashboardsvc/connections.go#L179 will understand it's an integration id and call `initUnknownConnection` for that integration.